### PR TITLE
Fix block images always add additional newlines on the end of the file

### PR DESCRIPTION
**Pull request by susnux**
### 📝 Summary

* Resolves: #3428

This fixes block images adding additional newlines, without this images at the end of a file will add two additional newlines.
This is fixed by using `closeBlock` within the `toMarkdown` function instead of always enforcing two newlines.

The tests are adjusted accordingly.

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/text/blob/master/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

File: src/tests/markdown.spec.js [hunk 1]
```diff
@@ -120,9 +105,11 @@ describe('Markdown though editor', () => {
 		expect(markdownThroughEditor('[bar\\\\]: /uri\n\n[bar\\\\]')).toBe('[bar\\\\](/uri)')
 	})
 	test('images', () => {
+		// Inline images
 		expect(markdownThroughEditor('text ![test](foo) moretext')).toBe('text ![test](foo) moretext')
-		// regression introduced in #3282. To be fixed in #3428.
-		expect(markdownThroughEditor('![test](foo)')).toBe('![test](foo)\n\n')
+		// regression introduced in #3282. See issue #3428.
```
Comment (max-nextcloud) [inline src/tests/markdown.spec.js:110]: This comment is outdated now :smiley: - we don't need it anymore.
```suggestion
```

**Review by max-nextcloud — APPROVED** [review]
Thanks a lot for addressing this. It fixes an annoying issue and the approach is much cleaner.

File: src/tests/markdown.spec.js (hunk 1 above)
Comment (susnux) [reply src/tests/markdown.spec.js:110]: I let the comment stay there to have some information why a test was added, but I can remove it as well :)

Comment (max-nextcloud) [reply src/tests/markdown.spec.js:110]: Okay... cool. I meant the comment as documenting a regression - now it's documenting a fixed regression - i guess that's fine. Either way it points to a relevant issue.

**Pull request merged**